lib/speech/speech_recognizer.ts

```diff
@@ -0,0 +1,233 @@
+// -*- mode: typescript; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of Genie
+//
+// Copyright 2021 The Board of Trustees of the Leland Stanford Junior University
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Giovanni Campagna <gcampagn@cs.stanford.edu>
+
+import * as Tp from 'thingpedia';
+import * as events from 'events';
+import * as Stream from 'stream';
+import WebSocket from 'ws';
+
+const URL = 'https://almond-nl.stanford.edu';
+
+class SpeechRequest extends Stream.Writable {
+    private _stream : Stream.Readable;
+    private _endDetected : boolean;
+    private _ended : boolean;
+    private _started : boolean;
+    private _vad : Tp.Capabilities.VadApi|null;
+    private consecutiveSilence : number;
+    private _bufferedMessages : Array<Buffer>;
+    private _piped : boolean;
+    private _endTimeout : NodeJS.Timeout|null = null;
+    private _connection : WebSocket|null = null;
+    private _listener : any;
```

Comment:
You can set `_listener` in the constructor and give it a function type instead of `any`